src/update_handler.c

```diff
@@ -2172,14 +2172,22 @@ static gboolean img_to_boot_emmc_handler(RaucImage *image, RaucSlot *dest_slot,
 				R_UPDATE_ERROR,
 				R_UPDATE_ERROR_FAILED,
 				"Can't resolve eMMC device %s", dest_slot->device);
-		goto out;
+		return FALSE;
+	}
+
+	if (g_str_has_suffix(realdev, "boot0") || g_str_has_suffix(realdev, "boot1")) {
+		g_set_error(error,
+				R_UPDATE_ERROR,
+				R_UPDATE_ERROR_FAILED,
+				"Device path '%s' refers to boot partition, but base device is expected", realdev);
+		return FALSE;
```

Comment:
We should check this during config loading. See for example `R_CONFIG_ERROR_INVALID_DEVICE` as used for `jffs2` slots.